MaxText/layers.py

```diff
@@ -328,40 +326,49 @@ class MultiHeadDotProductAttention(nn.Module):
 
 
   def apply_attention(self, query, key, value, enable_flash_attention,
-                      attention_bias, dropout_rng, deterministic):
+                      decoder_segment_ids, attention_bias, dropout_rng, deterministic, decode):
     """ Apply Attention
     """
-    if enable_flash_attention:
+    if enable_flash_attention and not decode:
```

Comment:
"If enable_flash_attention and decode" should be asserting, probably here if it is unsupported.

Reply:
Yes, decode is not supported with flash_attention